server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesUpdateProcessor.java

```diff
@@ -180,58 +180,33 @@ private void assertUserIsSender(MailboxSession session, MailAddress sender) thro
-    }
+        boolean targetContainsOutbox = targetMailboxes.stream().anyMatch(outboxMailboxes::contains);
+        boolean targetIsOnlyOutbox = targetMailboxes.stream().allMatch(outboxMailboxes::contains);
 
-    private void assertValidUpdate(List<Flags> futureFlags, boolean targetHasDraft, boolean targetHasNonDraft,
-                                   boolean targetIsOutbox, boolean originIsDraft) throws DraftMessageMailboxUpdateException {
-        assertMessageIsNotInDraftAndNonDraftMailboxes(targetHasDraft, targetHasNonDraft);
-        assertNoNonDraftMessageInsideDraftMailbox(futureFlags, targetHasDraft);
-        assertNoDraftMessageOutOfDraftMailbox(futureFlags, targetHasNonDraft);
-        assertOutboxMoveOnlyFromDraft(targetIsOutbox, originIsDraft);
+        assertOutboxMoveTargetsOnlyOutBox(targetContainsOutbox, targetIsOnlyOutbox);
+        assertOutboxMoveOnlyFromDraft(targetContainsOutbox, allMessagesWhereDrafts);
     }
 
-    private void assertOutboxMoveOnlyFromDraft(boolean targetIsOutbox, boolean originIsDraft) {
-        if (targetIsOutbox && !originIsDraft) {
-            throw new InvalidOutboxMoveException();
+    private void assertOutboxMoveTargetsOnlyOutBox(boolean targetContainsOutbox, boolean targetIsOnlyOutbox) {
+        if (targetContainsOutbox && !targetIsOnlyOutbox) {
+            throw new InvalidOutboxMoveException("When moving a message to Outbox, only Outboxes mailboxes should be targeted.");
         }
     }
 
-    private void assertMessageIsNotInDraftAndNonDraftMailboxes(boolean targetHasDraft, boolean targetHasNonDraft) throws DraftMessageMailboxUpdateException {
-        if (targetHasDraft && targetHasNonDraft) {
-            throw new DraftMessageMailboxUpdateException("One can not have a message in mailboxes that don't have all the `draft` role");
-        }
-    }
-
-    private void assertNoNonDraftMessageInsideDraftMailbox(List<Flags> futureFlags, boolean targetHasDraft) throws DraftMessageMailboxUpdateException {
-        if (targetHasDraft) {
-            boolean anyNonDraftMessage = futureFlags.stream().anyMatch(flags -> !flags.contains(Flags.Flag.DRAFT));
-            if (anyNonDraftMessage) {
-                throw new DraftMessageMailboxUpdateException("Messages without '$Draft' keyword are prohibited in drafts mailboxes");
-            }
-        }
-    }
-
-    private void assertNoDraftMessageOutOfDraftMailbox(List<Flags> futureFlags, boolean targetHasNonDraft) throws DraftMessageMailboxUpdateException {
-        if (targetHasNonDraft) {
-            boolean anyDraftMessage = futureFlags.stream().anyMatch(flags -> flags.contains(Flags.Flag.DRAFT));
-            if (anyDraftMessage) {
-                throw new DraftMessageMailboxUpdateException("Messages with '$Draft' keyword are prohibited in non drafts mailboxes");
-            }
+    private void assertOutboxMoveOnlyFromDraft(boolean targetIsOutbox, boolean allMessagesWhereDrafts) {
```

Comment:
method name is not ok

Reply:
I propose `assertOutboxMoveOriginallyHasDraftKeywordSet`